relayer/src/supervisor/client_state_filter.rs

```diff
@@ -0,0 +1,195 @@
+use std::collections::HashMap;
+use tracing::debug;
+
+use crate::object;
+use crate::registry::Registry;
+use anomaly::BoxError;
+use ibc::ics02_client::client_state::{AnyClientState, ClientState};
+use ibc::ics03_connection::connection::ConnectionEnd;
+use ibc::ics04_channel::error::Kind;
+use ibc::ics24_host::identifier::{ChainId, ChannelId, ClientId, ConnectionId, PortId};
+use ibc::Height;
+
+type Identifier = String;
+
+/// A cache storing filtering status (allow or deny) for
+/// arbitrary identifiers.
+#[derive(Default, Debug)]
+pub(crate) struct FilterCache {
```

Comment:
This struct is not strictly a cache since it contains the filtering logic as well, maybe rename it to `FilterPolicy` or something like that?

We can also rename the file/module to `filter_policy` or something.